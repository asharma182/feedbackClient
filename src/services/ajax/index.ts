export const createFeedbackForm = (data: any) => {
  let myHeaders = new Headers();
  myHeaders.append("Content-Type", "application/json");

  let requestOptions: any = {
    method: 'POST',
    headers: myHeaders,
    body: JSON.stringify(data),
    redirect: 'follow'
  };

  return fetch("http://localhost:3001/createFeedback", requestOptions)
    .then(response => response.text())
    .then(result => result)
    .catch(error => console.log('error', error));
}

export const getAllFeedbacks = () => {
  let requestOptions: any = {
    method: 'GET',
    redirect: 'follow'
  };

  return fetch("http://localhost:3001/feedback", requestOptions)
    .then(response => response.json())
    .then(result => result)
    .catch(error => console.log('error', error));

}

export const getOneFeedback = (id: number) => {
  let requestOptions: any = {
    method: 'GET',
    redirect: 'follow'
  };

  return fetch("http://localhost:3001/feedback/"+id, requestOptions)
    .then(response => response.json())
    .then(result => result)
    .catch(error => console.log('error', error));

}

export const removeFeedback = (id: number) => {

  var requestOptions: any = {
    method: 'DELETE',
    redirect: 'follow'
  };

  return fetch("http://localhost:3001/delete/feedback/" + id, requestOptions)
    .then(response => response.text())
    .then(result => console.log(result))
    .catch(error => console.log('error', error));
}

export const updateFeedback = (data: any) => {
  let myHeaders = new Headers();
  myHeaders.append("Content-Type", "application/json");

  var requestOptions: any = {
    method: 'PUT',
    headers: myHeaders,
    body: JSON.stringify(data),
    redirect: 'follow'
  };

  return fetch("http://localhost:3001/editFeedback/"+data.id, requestOptions)
    .then(response => response.text())
    .then(result => result )
    .catch(error => console.log('error', error));
}