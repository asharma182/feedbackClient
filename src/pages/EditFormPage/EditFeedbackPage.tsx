import React, { useEffect, useState } from 'react';
import { Form } from '../../component/Form'
import { useLocation, useHistory, useParams } from "react-router-dom"
import { InputField } from '../../component/InputField';
import { Textarea } from '../../component/Textarea';
import { getOneFeedback, updateFeedback } from '../../services/ajax';

interface Feedback {
    id: number,
    name: string,
    age: number,
    feedback: string
}

export const EditFeedbackPage = () => {
    const history = useHistory()
    const [feedback, setFeedback] = useState()
    const { id }: any = useParams();

    const handlePostEvent = () => {
        history.goBack();
    }
    useEffect(() => {
        async function getFeedback() {
            const getFeedbacks = await getOneFeedback(id)
            console.log(getFeedbacks)
            setFeedback(getFeedbacks[0])
        }
        getFeedback()
    }, [id])

    return (
        <Form onSubmit={updateFeedback} refresh={handlePostEvent} feedbackdata={feedback} />
    )
}