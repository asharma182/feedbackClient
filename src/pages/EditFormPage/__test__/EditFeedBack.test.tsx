import renderer from 'react-test-renderer';
import {EditFeedbackPage} from '../EditFeedbackPage';
import { getOneFeedback, updateFeedback } from '../../../services/ajax';
import React from 'react'
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // to use the real parts of react-router-dom for everything except the hooks in mocking.
    useParams: () => ({
      id: '1',
    }),
    useRouteMatch: () => ({ url: '/editFeedback/1' }),
  }));

describe('test for snapshot', () => {
it('renders correctly when there are no items', () => {
    // const spy = jest.spyOn(React, "useParams").mockImplementationOnce(
    //     () =>
    //       new Promise((resolve) =>
    //         resolve({
    //           title: "Hello World!",
    //           content: "My first post."
    //         })
    //       )
    //   );
    
    const tree = renderer.create(<EditFeedbackPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
