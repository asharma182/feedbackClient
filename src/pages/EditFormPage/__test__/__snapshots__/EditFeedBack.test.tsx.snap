// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for snapshot renders correctly when there are no items 1`] = `
<div
  className="form-cotainer"
  style={
    Object {
      "backgroundColor": "#f8f9fa",
      "height": "100%",
    }
  }
>
  <h5
    className="modal-title"
  >
    Provide us feedback
  </h5>
  <div
    className="modal-body"
  >
    <form
      id="contact-form"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="label"
          htmlFor="name"
        >
          Name
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="label"
          htmlFor="age"
        >
          Age
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
          value={0}
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="label"
          htmlFor="feedback"
        >
          Feedback
        </label>
        <textarea
          className="form-control"
          onChange={[Function]}
          rows={5}
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn button btn-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
